ZEngine/ZEngine/Rendering/Renderers/GraphicRenderer.cpp

```diff
@@ -292,28 +292,39 @@ namespace ZEngine::Rendering::Renderers
     {
-        unsigned char                        image_data[] = {255, 255, 255, 255, '\0'};
-
-        Specifications::TextureSpecification spec         = {
-                    .Width        = width,
-                    .Height       = height,
-                    .BytePerPixel = Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)],
-                    .Format       = Specifications::ImageFormat::R8G8B8A8_SRGB,
-                    .Data         = image_data,
+        size_t                               data_size = width * height * Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)];
+        std::vector<unsigned char>           image_data(data_size, 255);
+
+        Specifications::TextureSpecification spec = {
+            .Width        = width,
+            .Height       = height,
+            .BytePerPixel = Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)],
+            .Format       = Specifications::ImageFormat::R8G8B8A8_SRGB,
+            .Data         = image_data.data(),
         };
 
         return CreateTexture(spec);
     }
 
     Helpers::Ref<Textures::Texture> GraphicRenderer::CreateTexture(uint32_t width, uint32_t height, float r, float g, float b, float a)
     {
-        unsigned char image_data[]                = {0, 0, 0, 0, '\0'};
-        image_data[0]                             = static_cast<unsigned char>(std::clamp(r, .0f, 255.0f));
-        image_data[1]                             = static_cast<unsigned char>(std::clamp(g, .0f, 255.0f));
-        image_data[2]                             = static_cast<unsigned char>(std::clamp(b, .0f, 255.0f));
-        image_data[3]                             = static_cast<unsigned char>(std::clamp(a, .0f, 255.0f));
+        size_t                     data_size = width * height * Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)];
+        std::vector<unsigned char> image_data(data_size);
 
-        Specifications::TextureSpecification spec = {.Width = width, .Height = height, .BytePerPixel = Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)], .Format = Specifications::ImageFormat::R8G8B8A8_SRGB, .Data = image_data};
+        unsigned char              r_byte = static_cast<unsigned char>(std::clamp(r * 255.0f, 0.0f, 255.0f));
+        unsigned char              g_byte = static_cast<unsigned char>(std::clamp(g * 255.0f, 0.0f, 255.0f));
+        unsigned char              b_byte = static_cast<unsigned char>(std::clamp(b * 255.0f, 0.0f, 255.0f));
+        unsigned char              a_byte = static_cast<unsigned char>(std::clamp(a * 255.0f, 0.0f, 255.0f));
+
+        for (size_t i = 0; i < data_size; i += Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)])
```

Comment:
please simplify the expression, Specifications::BytePerChannelMap[VALUE_FROM_SPEC_MAP(Specifications::ImageFormat::R8G8B8A8_SRGB)] 
by making the call once at the top